src/Basis/Constructors.jl

```diff
@@ -310,6 +454,112 @@ function buildinterpolationandgradient(
+- from conformal maps following Hale and Trefethen (2008) SIAM J. NUMER. ANAL. Vol. 46, No. 2
+
+# Example:
+```jldoctest
+# get nodes, quadrature points, and weights
+numbernodes = 3;
+numberquadraturepoints = 4;
+nodes = LFAToolkit.lobattoquadrature(numbernodes, false);
+quadraturepoints, quadratureweights1d = LFAToolkit.gaussquadrature(numberquadraturepoints);
+
+# build interpolation, gradient matrices with transplanted quadrature from conformal maps
+interpolation, gradient = LFAToolkit.buildinterpolationandgradientmapped(nodes, quadraturepoints);
+
+# verify
+for i in 1:numberquadraturepoints
+    total = sum(interpolation[i, :]);
+    @assert total ≈ 1.0
+
+    total = sum(gradient[i, :]);
+    @assert abs(total) < 1e-14
+end
+
+# output
+
+```
+"""
+
+function buildinterpolationandgradientmapped(
+    nodes1d::AbstractArray{Float64},
+    quadraturepoints1d::AbstractArray{Float64},
+)
+    # check inputs
+    numbernodes1d = length(nodes1d)
+    numberquadratuepoints1d = length(quadraturepoints1d)
+    if numbernodes1d < 2
+        # COV_EXCL_START
+        throw(
+            DomanError(
+                numbernodes1d,
+                "length of nodes1d must be greater than or equal to 2",
+            ),
+        )
+        # COV_EXCL_STOP
+    end
+    if numbernodes1d < 2
+        # COV_EXCL_START
+        throw(
+            DomanError(
+                numbernodes1d,
+                "length of quadraturepoints1d must be greater than or equal to 2",
+            ),
+        )
+        # COV_EXCL_STOP
+    end
+
+    # build interpolation, gradient matrices from Fornberg, 1998
+    # and applying conformal maps resulting in transplanted quadrature
+    # following Hale and Trefethen (2008) SIAM J. NUMER. ANAL. Vol. 46, No. 2
+    interpolation1d = zeros(Float64, numberquadratuepoints1d, numbernodes1d)
+    gradient1d = zeros(Float64, numberquadratuepoints1d, numbernodes1d)
+    for i = 1:numberquadratuepoints1d
+        c1 = 1.0
+        c3 = nodes1d[1] - quadraturepoints1d[i]
+        interpolation1d[i, 1] = 1.0
+        for j = 2:numbernodes1d
+            c2 = 1.0
+            c4 = c3
+            c3 = nodes1d[j] - quadraturepoints1d[i]
+            for k = 1:j-1
+                dx = nodes1d[j] - nodes1d[k]
+                c2 *= dx
+                if k == j - 1
+                    # here we should multiply the gradient by inverse of g'
+                    gradient1d[i, j] =
+                        c1 * (interpolation1d[i, k] - c4 * gradient1d[i, k]) / c2
+                    interpolation1d[i, j] = -c1 * c4 * interpolation1d[i, k] / c2
+                end
+                # here we should multiply the gradient by the inverse of g'
+                gradient1d[i, k] = (c3 * gradient1d[i, k] - interpolation1d[i, k]) / dx
+                interpolation1d[i, k] = c3 * interpolation1d[i, k] / dx
+            end
+            c1 = c2
+        end
+    end
+
+    # return
+    return interpolation1d, gradient1d
+end
```

Comment:
I think this can be simpler. Suppose `x = g(X)` is the transformed point. Our basis functions are transformed polynomials so `\phi(x) = \phi(g(X)) = p(X)` pulls back as a polynomial, and thus are basis evaluator `B_{ij} = \phi_j(x_i) = p_j(X_i)` is exactly the polynomial matrix.

For derivatives, we have `d\phi(x)/dx = d p(X)/dX dX/dx = (dp/dX) / g'(X)`. So in the code, I think we can write
```julia
points, weights = gaussquadrature(Q)
nodes = lobattoquadrature(P)
B, D = buildinterpolationandgradient(nodes, points)
D = D ./ g_map_prime.(points) # scale each row
return B, D
```
This basis should pair with the mapped Lobotto nodes and the mapped Gauss quadrature.